simtools/production_configuration/derive_computing_resources.py

```diff
@@ -0,0 +1,207 @@
+                )
+                resources["storage_per_event"] = u.Quantity(
+                    resources["storage_per_event"]["value"], resources["storage_per_event"]["unit"]
+                )
+
+        return lookup
+
+    def estimate_resources(self) -> dict:
+        """
+        Estimate the compute and storage resources required for the simulation.
+
+        Returns
+        -------
+        dict
+            A dictionary with estimates for compute and storage resources, with units.
+        """
+        number_of_events = self.simulation_params.get("number_of_events")
+        if self.existing_data:
+            return self.interpolate_resources(number_of_events)
+
+        return self.guess_resources_per_event(number_of_events)
+
+    def interpolate_resources(self, number_of_events: int) -> dict:
+        """
+        Interpolate resources required for the simulation from existing data.
+
+        Parameters
+        ----------
+        number_of_events : int
+            The number of events for which to interpolate resources.
+
+        Returns
+        -------
+        dict
+            A dictionary with interpolated estimates for compute and storage resources, with units.
+        """
+        azimuth = self.grid_point["azimuth"]
+        zenith = self.grid_point["zenith"]
+        nsb = self.grid_point["night_sky_background"]
+
+        closest_data = min(
+            self.existing_data,
+            key=lambda x: (
+                abs(x["azimuth"] - azimuth) + abs(x["zenith"] - zenith) + abs(x["nsb"] - nsb)
+            ),
+        )
+
+        compute_total_value = closest_data["compute_total"]["value"]
+        compute_total_unit = closest_data["compute_total"]["unit"]
+
+        storage_total_value = closest_data["storage_total"]["value"]
+        storage_total_unit = closest_data["storage_total"]["unit"]
+
+        compute_total = (
+            compute_total_value
+            * (number_of_events / closest_data["events"])
+            * u.Unit(compute_total_unit)
+        )
+        storage_total = (
+            storage_total_value
+            * (number_of_events / closest_data["events"])
+            * u.Unit(storage_total_unit)
+        )
+
+        return {"compute_total": compute_total, "storage_total": storage_total}
+
+    def guess_resources_per_event(self, number_of_events: int) -> dict:
+        """
+        Estimate resources for the simulation based on grid point parameters and per-event guess.
+
+        Parameters
+        ----------
+        number_of_events : int
+            The number of events for which to estimate resources.
+
+        Returns
+        -------
+        dict
+            A dictionary with guessed estimates for compute and storage resources, with units.
+        """
+        zenith = self.grid_point["zenith"]
+        zeniths = sorted(self.lookup_table[self.site]["Zenith"].keys())
+
+        if zenith <= zeniths[0]:
```

Comment:
Are we sure that we want to allow `<` or `>` in this context? This might mean that resource estimates are off from the actual needs if the zenith angle is very different from those available in the table. I would suggest not to allow that and issue an exception?